src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/format-hex/Format-Hex.cs

```diff
@@ -183,84 +222,148 @@ private void ProcessFileContent(string path)
-                        {
-                            bytesRead -= (int)(count - Count);
-                            WriteHexadecimal(buffer.Slice(0, bytesRead), path, offset);
-                            break;
-                        }
-
+                        bytesRead -= (int)(count - Count);
                         WriteHexadecimal(buffer.Slice(0, bytesRead), path, offset);
-
-                        offset += bytesRead;
+                        break;
                     }
+
+                    WriteHexadecimal(buffer.Slice(0, bytesRead), path, offset);
+
+                    offset += bytesRead;
                 }
             }
-            catch (IOException ioException)
+            catch (IOException fileException)
             {
                 // IOException takes care of FileNotFoundException, DirectoryNotFoundException, and PathTooLongException
-                WriteError(new ErrorRecord(ioException, "FormatHexIOError", ErrorCategory.WriteError, path));
+                WriteError(new ErrorRecord(fileException, "FormatHexIOError", ErrorCategory.WriteError, path));
             }
             catch (ArgumentException argException)
             {
                 WriteError(new ErrorRecord(argException, "FormatHexArgumentError", ErrorCategory.WriteError, path));
             }
             catch (NotSupportedException notSupportedException)
             {
-                WriteError(new ErrorRecord(notSupportedException, "FormatHexPathRefersToANonFileDevice", ErrorCategory.InvalidArgument, path));
+                WriteError(new ErrorRecord(
+                    notSupportedException,
+                    "FormatHexPathRefersToANonFileDevice",
+                    ErrorCategory.InvalidArgument,
+                    path));
             }
             catch (SecurityException securityException)
             {
-                WriteError(new ErrorRecord(securityException, "FormatHexUnauthorizedAccessError", ErrorCategory.PermissionDenied, path));
+                WriteError(new ErrorRecord(
+                    securityException,
+                    "FormatHexUnauthorizedAccessError",
+                    ErrorCategory.PermissionDenied,
+                    path));
             }
         }
 
         #endregion
 
         #region InputObjects
 
+        private void ProcessString(string originalString)
+        {
+            Span<byte> bytes = Encoding.GetBytes(originalString);
+
+            int offset = Math.Min(bytes.Length, Offset < int.MaxValue ? (int)Offset : int.MaxValue);
+            int count = Math.Min(bytes.Length - offset, Count < int.MaxValue ? (int)Count : int.MaxValue);
+
+            if (offset != 0 || count != bytes.Length)
+            {
+                WriteHexadecimal(bytes.Slice(offset, count), offset: 0, GetGroupLabel(typeof(string)));
+            }
+            else
+            {
+                WriteHexadecimal(bytes, offset: 0, label: GetGroupLabel(typeof(string)));
```

Comment:
Should either consistently name the label param or don't, but it's a bit inconsistent.

Reply:
I think I'll opt for naming it, since one of the other possible overloads _also_ has a string param (`path`), so it's safer to name this one.